go/pkg/services/logcollector/statefile.go

```diff
@@ -0,0 +1,174 @@
+// Copyright 2018 Iguazio
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package logcollector
```

Comment:
move abstraction to `logcollector/statestore` and there have
1. types.go with interface definition
2. file `/file/client.go` that implements
3. file `/mock/client.go` that implements mock (not needed on this specific PR, unless you need for tests
4. factory that returns upon kind